src/content/1.7/modules/sample_modules/order_view_page_new_hooks/view-order-hooks-signature-widget.md

```diff
@@ -0,0 +1,448 @@
+            'filename' => $this->signatureImgDir.$signature->getFilename()
+        ];
+    }
+}
+```
+Then lets use Symfony Dependency Injection 
+(https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/).
+and create services configuration for the above 
+classes in `config/services.yml`. For more information: 
+https://devdocs.prestashop.com/1.7/modules/concepts/services/#symfony-services
+The intent behind dependency injection is to achieve Separation of Concerns of construction and use of objects. 
+This can increase readability and code reuse, reduce dependencies, lead to more testable code.
+```yaml
+parameters:
+  signatureImgDirectory: 'signatures/'
+
+services:
+  prestashop.module.demovieworderhooks:
+    class: DemoViewOrderHooks
+    factory: [Module, getInstanceByName]
+    arguments:
+      - 'demovieworderhooks'
+
+  prestashop.module.demovieworderhooks.repository.order_repository:
+    class: PrestaShop\Module\DemoViewOrderHooks\Repository\OrderRepository
+
+  prestashop.module.demovieworderhooks.repository.order_signature_repository:
+    class: PrestaShop\Module\DemoViewOrderHooks\Repository\OrderSignatureRepository
+    factory: ['@doctrine.orm.default_entity_manager', getRepository]
+    arguments:
+      - PrestaShop\Module\DemoViewOrderHooks\Entity\OrderSignature
+
+  prestashop.module.demovieworderhooks.presenter.order_signature_presenter:
+    class: PrestaShop\Module\DemoViewOrderHooks\Presenter\OrderSignaturePresenter
+    arguments:
+      - '@=service("prestashop.module.demovieworderhooks").getPathUri() ~ parameter("signatureImgDirectory")'
+
+```
+
+Let's create a twig templates in `modules/demovieworderhooks/views/templates/admin/`:
+
+card.html.twig
+
+```twig
+{% trans_default_domain 'Module.Demovieworderhooks.Admin' %}
+
+<div class="card">
+  <div class="card-header">
+    <h3 class="card-header-title">
+      {% block card_title %}
+      {% endblock %}
+    </h3>
+  </div>
+
+  <div class="card-body">
+    {% block card_body %}
+    {% endblock %}
+  </div>
+</div>
+```
+
+`customer_signature.html.twig` extending `card.html.twig`
+
+```twig
+{% extends '@Modules/demovieworderhooks/views/templates/admin/card.html.twig' %}
+
+{% block card_title %}
+  {{ 'Signature'|trans }}
+{% endblock %}
+
+{% block card_body %}
+  <div class="col-lg">
+    <div class="display-4 text-center">
+      {{ signature.gender }} {{ signature.firstName }} {{ signature.lastName }}
+    </div>
+    <div class="text-center">
+      <img src="{{ signature.imagePath }}" alt="">
+    </div>
+  </div>
+{% endblock %}
+```
+
+Lets add several methods to `DemoViewOrderHooks` class.
+`getModuleTemplatePath` - get's the path of the templates folder.
+
+```php
+    /**
+     * Get path to this module's template directory
+     */
+    private function getModuleTemplatePath(): string
+    {
+        return "@Modules/$this->name/views/templates/admin/";
```

Comment:
We shouldn't recommend using variables inside string, but the concatenation is a bit "heavy" as well Maybe we can use a `sprintf()` like we often do for exception messages in the core?